x/mongo/driver/topology/polling_srv_records_test.go

```diff
@@ -281,16 +281,48 @@ func TestPollSRVRecords(t *testing.T) {
 		}
 		_ = topo.Disconnect(context.Background())
 	})
-	t.Run("polling is not required for load balanced clusters", func(t *testing.T) {
-		cs, err := connstring.ParseAndValidate("mongodb+srv://test1.test.build.10gen.cc/?heartbeatFrequencyMS=100")
-		assert.Nil(t, err, "connstring.ParseAndValidate error: %v", err)
+}
+
+func TestPollingSRVRecordsLoadBalanced(t *testing.T) {
+	createLBTopology := func(t *testing.T, uri string) *Topology {
+		t.Helper()
 
+		cs, err := connstring.ParseAndValidate(uri)
+		assert.Nil(t, err, "connstring.ParseAndValidate error: %v", err)
 		topo, err := New(
 			WithConnString(func(connstring.ConnString) connstring.ConnString { return cs }),
 			WithURI(func(string) string { return cs.Original }),
 			WithLoadBalanced(func(bool) bool { return true }),
 		)
 		assert.Nil(t, err, "topology.New error: %v", err)
+		return topo
+	}
+
+	t.Run("pollingRequired is set to false", func(t *testing.T) {
+		topo := createLBTopology(t, "mongodb+srv://test1.test.build.10gen.cc/?heartbeatFrequencyMS=100")
 		assert.False(t, topo.pollingRequired, "expected SRV polling to not be required, but it is")
 	})
+
+	t.Run("new records are not detected", func(t *testing.T) {
+		recordsToAdd := []*net.SRV{{"localhost.test.build.10gen.cc.", 27019, 0, 0}}
+		mockResolver := newMockResolver(recordsToAdd, nil, false, false)
+		dnsResolver := &dns.Resolver{
+			LookupSRV: mockResolver.LookupSRV,
+			LookupTXT: mockResolver.LookupTXT,
+		}
+
+		topo := createLBTopology(t, "mongodb+srv://test3.test.build.10gen.cc/?loadBalanced=true")
```

Comment:
Nit: is `?loadBalanced=true` redundant here since `createLBTopology` sets the topology's loadbalanced option?

Reply:
Good catch. Removed. I also realized that `createLBTopology` is not setting the appropriate server/connection options to enable LB mode at all levels. This probably isn't relevant for this specific test, but I refactored it a bit to set `LoadBalanced` on the parsed `ConnString` instead so those options are all handled by `WithConnString`.